lib/drivers/weitai/weiTaiUSB.js

```diff
@@ -0,0 +1,1565 @@
+            alarmRecord,
+            new Date(alarmdatum.deviceTime).valueOf()
+          );
+          alarmRecord.done();
+        } catch (e) {
+          debug('alarm', e);
+        }
+
+        alermRes.push(alarmRecord);
+      }
+      return alermRes;
+    },
+    formatDecimal(originnum, decimal) {
+      let num = originnum.toString();
+      let index = num.indexOf('.');
+      let resNum = '';
+      if (index !== -1) {
+        resNum = num.substring(0, decimal + index + 1);
+      } else {
+        resNum = num.substring(0);
+      }
+      return parseFloat(resNum).toFixed(decimal);
+    },
+    fetchData(progress, data, cb) {
+      // debug('in fetchData', data);
+      let records = [];
+      let incomingA = data.incomingA;
+      const inComeRes = WeitaiUSB.parsePacket(incomingA, cfg, cb); //inComeRes:{BloodGlucoses:[],BasalRates:[],BolusRates:[]}
+      data.BloodGlucoses = inComeRes.BloodGlucoses;
+      data.BasalRates = inComeRes.BasalRates;
+      data.BolusRates = inComeRes.BolusRates;
+      data.alarm = inComeRes.alarm;
+      data.records = records;
+      return cb(null, data);
+    },
+
+    processData(progress, data, cb) {
+      progress(100);
+      let settings = data.settings;
+      let alerms = data.alarm;
+      let postSetting = this.buildSettings(settings);
+      let bloodRes = this.buildBlood(data.BloodGlucoses);
+      let basalRes = this.buildBasal(data.BasalRates);
+      // let basalRes = this.buildBasal(testBasal);
+      let bolusRes = this.buildBolus(data.BolusRates);
+      // let bolusRes = this.buildBolus(testArray);
+      basalRes = basalRes.length > 1 ? basalRes : [];
+      let alermRes = this.buildAlerm(alerms);
+      let post_records = [].concat(
+        bloodRes,
+        basalRes,
+        bolusRes,
+        postSetting,
+        alermRes
+      );
+      if (!post_records.length) {
+        return cb({ code: 'E_WEITAI_NO_UPLOAD' }, null);
+      }
+      data.post_records = post_records;
+      return cb(null, data);
+    },
+
+    uploadData(progress, data, cb) {
+      progress(0);
+      const sessionInfo = {
+        delta: cfg.delta,
+        deviceTags: cfg.deviceInfo.tags,
+        // deviceManufacturers: cfg.deviceInfo.manufacturers,
+        // deviceModel: cfg.deviceInfo.model,
+        deviceManufacturers: ['Weitai'],
+        deviceModel: 'weitaiupload',
+        deviceSerialNumber: cfg.deviceInfo.serialNumber || 'sdsdgdfdg',
+        deviceId: cfg.deviceInfo.deviceId || 'MTM-C2',
+        start: sundial.utcDateString(),
+        timeProcessing: cfg.tzoUtil.type,
+        tzName: cfg.timezone,
+        version: cfg.version,
+      };
+      // return false;
+      cfg.api.upload.toPlatform(
+        data.post_records,
+        sessionInfo,
+        progress,
+        cfg.groupId,
+        (err, result) => {
+          progress(100);
+
+          if (err) {
+            debug(err);
+            debug(result);
+            cb(err, data);
```

Comment:
```suggestion
            return cb(err, data);
```

If you don't return here, it will callback twice.